pkg/skaffold/deploy/helm/args.go

```diff
@@ -201,3 +204,54 @@ func (h *Deployer) installArgs(r latest.HelmRelease, builds []build.Artifact, va
 
 	return args, nil
 }
+
+// sanitizeFilePath is used to sanitize filepaths that are provided to the `setFiles` flag
+// helm `setFiles` doesn't work with the unescaped filepath separator (\) for Windows or if there are unescaped tabs and spaces in the directory names.
+// So we escape all odd count occurrences of `\` for Windows, and wrap the entire string in quotes if it has spaces.
+func sanitizeFilePath(s string) string {
+	needsQuotes := false
+	if len(s) == 0 {
+		return `""`
+	}
+
+	for i := 0; i < len(s); i++ {
+		switch s[i] {
+		case ' ', '\t':
+			needsQuotes = true
+			break
+		}
+	}
+
+	if runtime.GOOS != "windows" {
+		if needsQuotes {
+			return fmt.Sprintf(`"%s"`, s)
```

Comment:
What about paths with embedded quotes? 

(It feels like there should be a library out there that does this!)

Reply:
can a filepath have embedded quotes? Currently the only client of this is the `setFiles` parameter for the `helm` deployer and the yaml parser doesn't preserve wrapping quotes anyways.

Reply:
I think erroring out with the `helm` propagated error message is enough in these scenarios?

Reply:
Ok — I wrote the above thinking that we needed to do escaping for `CMD /C`, not Helm.